services/ui_backend_service/api/autocomplete.py

```diff
@@ -0,0 +1,110 @@
+from services.utils import handle_exceptions, web_response
+import asyncio
+import threading
+
+
+class AutoCompleteApi(object):
+    # Cache tags so we don't have to request DB everytime
+    tags = []
+
+    def __init__(self, app, db):
+        self.db = db
+        # Cached resources
+        app.router.add_route("GET", "/autocomplete/tags", self.get_all_tags)
+        # Non-cached resources
+        app.router.add_route("GET", "/autocomplete/flows", self.get_all_flows)
+        app.router.add_route("GET", "/autocomplete/flows/{flow_id}/runs", self.get_runs_for_flow)
+        app.router.add_route("GET", "/autocomplete/flows/{flow_id}/runs/{run_id}/steps", self.get_steps_for_run)
+        self._async_table = self.db.run_table_postgres
+        self.loop = asyncio.get_event_loop()
+        self.loop.create_task(self.setup_tags())
```

Comment:
recursively continuing tasks works as well, but a simpler solution might be to have a poller task that uses asyncio.sleep at the end? see `heartbeat_monitor.py#check_heartbeats` for an example.